include/zephyr/drivers/gpio.h

```diff
@@ -1230,13 +1242,26 @@ static inline int gpio_pin_get_config_dt(const struct gpio_dt_spec *spec,
 __syscall int gpio_port_get_raw(const struct device *port,
 				gpio_port_value_t *value);
 
-static inline int z_impl_gpio_port_get_raw(const struct device *port,
-					   gpio_port_value_t *value)
+static inline int z_impl_gpio_port_get_raw(const struct device *port, gpio_port_value_t *value)
 {
-	const struct gpio_driver_api *api =
-		(const struct gpio_driver_api *)port->api;
+	const struct gpio_driver_api *api = (const struct gpio_driver_api *)port->api;
+
+	int ret = api->port_get_raw(port, value);
+
+	if (ret != 0) {
+		return ret;
+	}
+
+	gpio_pin_t counter = 0;
 
-	return api->port_get_raw(port, value);
+	for (gpio_port_value_t val_cp = *value; IS_ENABLED(CONFIG_TRACING_GPIO) && (val_cp != 0);
+	     val_cp >>= 1, counter++) {
+		if ((val_cp & 0x1) == 1) {
+			sys_port_trace_gpio_pin_active(port, counter);
+		}
+	}
```

Comment:
Why not add tracing for `port_get_raw()`? this conversion from mask to individual pin events will likely take even longer than the call to the API itself

Reply:
also, it does not seem to respect the ACTIVE_HIGH/ACTIVE_LOW flags, which determine if a 1 or 0 means `active` or not, which it seems is what is being traced by `sys_port_trace_gpio_pin_active()` rather than the raw value.

Reply:
This should be addressed. Please unresolve if you think it is not.